Failed to build with GHCJS

I'm trying to build the project that has `network` library as a dependency using GHCJS compiler, but, unfortunately, I get the following error:

```haskell
Failed to build network-3.1.0.1.
Build log (
/home/vrom911/.cabal/logs/ghcjs-8.4.0.1/network-3.1.0.1-c2dd67eaf943e0b241f84c699a37a0c8bfce83a37ecc86e20e87d1023fb261cd.log
):
Configuring network-3.1.0.1...
configure: WARNING: unrecognized options: --with-compiler
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for gcc... /usr/bin/gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /usr/bin/gcc accepts -g... yes
checking for /usr/bin/gcc option to accept ISO C89... none needed
checking for an ANSI C-conforming const... yes
checking how to run the C preprocessor... /usr/bin/gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/types.h... (cached) yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sys/uio.h usability... yes
checking sys/uio.h presence... yes
checking for sys/uio.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking netinet/tcp.h usability... yes
checking netinet/tcp.h presence... yes
checking for netinet/tcp.h... yes
checking sys/un.h usability... yes
checking sys/un.h presence... yes
checking for sys/un.h... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking net/if.h usability... yes
checking net/if.h presence... yes
checking for net/if.h... yes
checking netioapi.h usability... no
checking netioapi.h presence... no
checking for netioapi.h... no
checking for struct ucred... yes
checking for gai_strerror... yes
checking for gethostent... yes
checking for accept4... yes
checking for getpeereid... no
checking whether AI_ADDRCONFIG is declared... yes
checking whether AI_ALL is declared... yes
checking whether AI_NUMERICSERV is declared... yes
checking whether AI_V4MAPPED is declared... yes
checking whether IPV6_V6ONLY is declared... yes
checking whether IPPROTO_IP is declared... yes
checking whether IPPROTO_TCP is declared... yes
checking whether IPPROTO_IPV6 is declared... yes
checking whether SO_PEERCRED is declared... yes
checking for struct msghdr.msg_control... yes
checking for struct msghdr.msg_accrights... no
checking for struct sockaddr.sa_len... no
configure: creating ./network.buildinfo
configure: creating ./config.status
config.status: creating include/HsNetworkConfig.h
configure: WARNING: unrecognized options: --with-compiler
Preprocessing library for network-3.1.0.1..
Building library for network-3.1.0.1..
[ 1 of 24] Compiling Network.Socket.Imports ( Network/Socket/Imports.hs, dist/build/Network/Socket/Imports.js_o )
[ 2 of 24] Compiling Network.Socket.If ( Network/Socket/If.hs, dist/build/Network/Socket/If.js_o )
[ 3 of 24] Compiling Network.Socket.Cbits ( dist/build/Network/Socket/Cbits.hs, dist/build/Network/Socket/Cbits.js_o )
[ 4 of 24] Compiling Network.Socket.Fcntl ( Network/Socket/Fcntl.hs, dist/build/Network/Socket/Fcntl.js_o )

Network/Socket/Fcntl.hs:26:29: error:
    Not in scope: ‘System.Posix.Internals.setCloseOnExec’
    Perhaps you meant ‘System.Posix.Internals.setCooked’ (imported from System.Posix.Internals)
    Module ‘System.Posix.Internals’ does not export ‘setCloseOnExec’.
   |
26 | setCloseOnExecIfNeeded fd = System.Posix.Internals.setCloseOnExec fd
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build network-3.1.0.1 (which is required by exe:issue-wanted
from issue-wanted-0.0.0.0 and exe:frontend-exe from frontend-0.0.0.0). See the
build log above for details.

```

Could you help me to fix this?

I'm using `ghcjs-8.4`. building with `cabal-install version 3.1.0.0`.

--- a/Network/Socket/Fcntl.hs
+++ b/Network/Socket/Fcntl.hs
@@ -20,7 +20,7 @@ setNonBlockIfNeeded fd =
 --
 --   Since 2.7.0.0.
 setCloseOnExecIfNeeded :: CInt -> IO ()
-#if defined(mingw32_HOST_OS)
+#if defined(mingw32_HOST_OS) || defined(ghcjs_HOST_OS)
 setCloseOnExecIfNeeded _ = return ()
 #else
 setCloseOnExecIfNeeded fd = System.Posix.Internals.setCloseOnExec fd
@@ -36,7 +36,7 @@ foreign import ccall unsafe "fcntl"
 --
 --   Since 2.7.0.0.
 getCloseOnExec :: CInt -> IO Bool
-#if defined(mingw32_HOST_OS)
+#if defined(mingw32_HOST_OS) || defined(ghcjs_HOST_OS)
 getCloseOnExec _ = return False
 #else
 getCloseOnExec fd = do

